broker/src/serve_tasks.rs

```diff
@@ -99,6 +149,73 @@ fn wait_get_statuscode<S>(vec: &Vec<S>, block: &HowLongToBlock) -> StatusCode {
+-> impl Stream<Item = Result<Event,Infallible>> + 'a
+where M: Clone + HasWaitId<I>
+{
+    let wait_until =
+        time::Instant::now() + block.wait_time.unwrap_or(time::Duration::from_secs(31536000));
+    trace!(
+        "Now is {:?}. Will wait until {:?}",
+        time::Instant::now(),
+        wait_until
+    );
+    let mut running = true;
+    let stream = async_stream::stream! {
+    while usize::from(block.wait_count.unwrap_or(0)) > results.len()
+        && time::Instant::now() < wait_until
+        && running {
+        trace!(
+            "Items in vec: {}, time remaining: {:?}",
+            results.len(),
+            wait_until - time::Instant::now()
+        );
+            tokio::select! {
+                _ = tokio::time::sleep_until(wait_until) => {
+                    debug!("SSE: Wait expired.");
+                    yield Ok(Event::default()
+                        .event(SseEventType::WaitExpired)
+                        .data("{}"));
+                    running = false;
+                },
+                result = new_result_rx.recv() => {
+                    match result {
+                        Ok(req) => {
+                            if filter.matches(&req) {
+                                let previous = results.insert(req.get_from().clone(), req.clone());
+                                let event_type = match previous {
+                                    Some(_) => SseEventType::UpdatedResult,
+                                    None => SseEventType::NewResult
+                                };
+                                yield Ok(Event::default()
+                                    .event(event_type)
+                                    .json_data(&req)
+                                    .unwrap());
+                            }
+                        },
+                        Err(e) => { panic!("Unable to receive from queue new_result_rx: {}", e); }
+                    }
+                },
+                deleted_task_id = deleted_task_rx.recv() => {
+                    match deleted_task_id {
+                        Ok(deleted_task_id) => {
+                            if deleted_task_id == *task_id {
+                                warn!("Task {} was just deleted while someone was waiting for results. Returning the {} results up to now.", task_id, results.len());
+                                yield Ok(Event::default()
+                                    .event(SseEventType::DeletedTask)
+                                    .data("{ \"task_id\": \"{deleted_task_id}\" }"));
+                                running = false;
+                            }
+                        },
+                        Err(e) => { panic!("Unable to receive from queue deleted_task_rx: {}", e); }
```

Comment:
Same as above, is this really globally non recoverable?

Reply:
see above